# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1424327454 18000
#      Thu Feb 19 01:30:54 2015 -0500
# Node ID ed1bafbb87ee7d3373a77f862c060edd0ab743e7
# Parent  a5206dfbaa1355c0fdf83a9672c1a2de386b083e
allow mouse wheel zoom amount to be configurable for qt graphics (bug #44304)

* Canvas.cc (Canvas::canvasWheelEvent): Use mousewheelzoom axes
property to compute zoom factor.

diff --git a/libgui/graphics/Canvas.cc b/libgui/graphics/Canvas.cc
--- a/libgui/graphics/Canvas.cc
+++ b/libgui/graphics/Canvas.cc
@@ -683,17 +683,23 @@ void Canvas::canvasWheelEvent (QWheelEve
           bool redrawFigure = true;
 
           switch (newMouseMode)
             {
             case ZoomMode:
               {
                 axes::properties& ap = Utils::properties<axes> (axesObj);
 
-                double factor = event->delta () > 0 ? 2.0 : 0.5;
+                // Control how fast to zoom when using scroll wheel.
+                double wheel_zoom_speed = ap.get_mousewheelzoom ();
+
+                // Determine if we're zooming in or out.
+                double factor = (event->delta () > 0
+                                 ? 1 / (1.0 - wheel_zoom_speed)
+                                 : 1.0 - wheel_zoom_speed);
 
                 ap.zoom (mode, factor);
 
 #if 0
                 Matrix view = ap.get_view ().matrix_value ();
                 if (view(1) != 90)
                   {
                     Matrix zlimits = ap.get_zlim ().matrix_value ();
